File: KisaragiTech.Dape/User/Service/TentativeUserFactory.cs
using KisaragiTech.Dape.User.Model;

namespace KisaragiTech.Dape.User.Service;

public static class TentativeUserFactory
{
    public static LocalTentativeUser Create(string initialPreferredUsername)
    {
        return new LocalTentativeUser(UserIDGenerationService.Generate(), initialPreferredUsername);
    }
}

File: src/main/scala/io/github/ksrgtech/dape/user/service/TentativeUserFactory.scala
package io.github.ksrgtech.dape.user.service

import cats.effect.kernel.Sync
import cats.implicits.*
import io.github.ksrgtech.dape.user.model.LocalTentativeUser

object TentativeUserFactory {

  def create[F[_]: Sync](initialPreferredUsername: String): F[LocalTentativeUser] =
    UserIdGenerationService.generate[F].map { userId =>
      new LocalTentativeUser(userId, initialPreferredUsername)
    }
}
